src/sql/platform/queryHistory/common/queryHistoryServiceImpl.ts

```diff
@@ -0,0 +1,84 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IQueryHistoryService } from 'sql/platform/queryHistory/common/queryHistoryService.ts';
+
+import { IQueryModelService, IQueryEvent } from 'sql/platform/query/common/queryModel';
+import { IModelService } from 'vs/editor/common/services/modelService';
+import { URI } from 'vs/base/common/uri';
+import { Range } from 'vs/editor/common/core/range';
+import { QueryHistoryInfo, QueryStatus } from 'sql/platform/queryHistory/common/queryHistoryInfo';
+import { IConnectionManagementService } from 'sql/platform/connection/common/connectionManagement';
+import { Event, Emitter } from 'vs/base/common/event';
+
+/**
+ * Service that collects the results of executed queries
+ */
+export class QueryHistoryService implements IQueryHistoryService {
+	_serviceBrand: any;
+
+	// MEMBER VARIABLES ////////////////////////////////////////////////////
+	private _infos: QueryHistoryInfo[] = [];
+	private _onInfosUpdated: Emitter<QueryHistoryInfo[]> = new Emitter<QueryHistoryInfo[]>();
+
+	// EVENTS //////////////////////////////////////////////////////////////
+	public get onInfosUpdated(): Event<QueryHistoryInfo[]> { return this._onInfosUpdated.event; }
+
+	// CONSTRUCTOR /////////////////////////////////////////////////////////
+	constructor(
+		@IQueryModelService _queryModelService: IQueryModelService,
+		@IModelService _modelService: IModelService,
+		@IConnectionManagementService _connectionManagementService: IConnectionManagementService
+	) {
+		_queryModelService.onQueryEvent((e: IQueryEvent) => {
```

Comment:
I think we register these as disposable normally? I know this might be different in services since they're intended to run for the length of the app, but would be good to see what we do elsewhere.